### ISSUE TITLE
V2 - CSV Export - CSV is blank

### ISSUE THREAD

**Issue by JCAAUSFRA** [Microsoft/VoTT#797]
**Describe the bug**

**To Reproduce**

**Expected behavior**

**Screenshots**

**Desktop (please complete the following information):**

**Additional context**

**Comment by wbreza** [comment]
Looks like there is one line of data in the csv from the screenshot. How many regions were labeled in the dataset?

**Comment by JCAAUSFRA** [comment]
Correct. Only one line was extracted in the data. Total number should be 1100ish

**Comment by wbreza** [comment]
@JCAAUSFRA - Are you source assets linked from local file system assets? or a cloud source?  Can you share a minimal example that reproduces?  I was not able to reproduce with our test data sets.

**Comment by JCAAUSFRA** [comment]
Local asset. But it is a server on our internal network, not on my personal computer.

**Comment by JCAAUSFRA** [comment]
I played around with the export functions and have found some interesting information that might get to the bottom of the issue. 

In order to explain in I need to give you the timeline:
1. started a new project
2. Tagged around 1000 images
3. Changed the export location and export format to CSV from JSON
4. Tried to export project in CSV and got nothing
5. Visited another image and tagged a new image.
6. Exported the project in CSV. Got one image
7. visited and tagged another 5 images
8. Exported the project in CSV and got all 6 images exported (1 from previous and 5 from current round)

 It is like the VoTT export system is only referencing the images I visited after the export format change rather then going back and exporting ALL previously visited and tagged images. 

Hope this explanation helps and sheds some light.

**Comment by JCAAUSFRA** [comment]
Had more of a play around. I decided if the exporter just needs me to revisit the images then I could just scroll back through all the images and the CSV file would populate. 

However this did not occur. It seemed to re-write the file to no detection.

**Comment by wbreza** [comment]
@JCAAUSFRA - The export will always restart and process all images in your project based on your export settings.  Which option are you choosing?  All assets, visited? or only tagged assets?  

If you are able to reproduce a smaller use case it will be easier to diagnose the issue.  Feel free to email me - wallace.breza@microsoft.com

**Comment by xzessmedia** [comment]
this is really annoying, propably using vott is not the best solution right now but maybe im missing something here:

> To Reproduce
Steps to reproduce the behavior:

    Label Data
    Ensure export is in CSV
    click the export button

first off  all, there is no export button, only save export settings.
this does not generate any csv.
by using the electron v2 version, i only get the vott json format. i can not export any other than that even if i select csv or something other in export settings.

also in the description it says export to yolo but there is literally no information on that topic anywhere.
all git issues with the topic "yolo" has been closed.. are those closed issues fixed? it does not seem to.
why are those issues closed without any information about the result?

**so please tell me how to export vott data to darknet / yolo ? now lets see if anybody can answer this!**

**Comment by xzessmedia** [comment]
> 
> 
> @JCAAUSFRA - The export will always restart and process all images in your project based on your export settings. Which option are you choosing? All assets, visited? or only tagged assets?
> 
> If you are able to reproduce a smaller use case it will be easier to diagnose the issue. Feel free to email me - [wallace.breza@microsoft.com](mailto:wallace.breza@microsoft.com)

i can not confirm even that!
maybe you should mark this piece of software with the "experimental" attribute

**Comment by tbarlow12** [comment]
@xzessmedia feel free to post another issue or better yet, open a pull request implementing the yolo export provider. This thread is regarding the CSV exporter

**Comment by JCAAUSFRA** [comment]
Hi @wbreza,

I managed to reproduce it on another project. 

Essentially if you change the export file type (CSV to VOTT JASON, for example), it seems to reset, ignoring all the previous tags you may have done. 

Would there be a way to for the previous work to be held in a "reference" file format before it is exported. That way if we change between the export types it doesn't require us to start all over again?

**Comment by pranavshukla1011** [comment]
I did it and had the same problem.
Try tagging a single image and saving it simultaneously(ctrl+S).

Its a bit tedious but worked for me.
